packages/beacon-node/src/api/impl/validator/index.ts

```diff
@@ -541,7 +557,10 @@ export function getValidatorApi({
       if (fullBlock && blindedBlock) {
         switch (builderSelection) {
           case routes.validator.BuilderSelection.MaxProfit: {
-            if (blockValueEngine >= blockValueBuilder) {
+            if (
+              builderBoostFactor === MAX_BUILDER_BOOST_FACTOR ||
+              blockValueEngine >= (blockValueBuilder * builderBoostFactor) / BigInt(100)
+            ) {
```

Comment:
I think `case routes.validator.BuilderSelection.ExecutionOnly` is no longer reachable in this switch statement. 
`builderSelection === ExecutionOnly` yields `blindedBlockPromise` to be null. Which means `blindedBlock` will always be null too.

Reply:
`builderSelection` is deprecated and will be removed as you noted that bootFactor takes care of it, so let the conditions be clear here till the followup PR removes it